marshmallow_recipe/fields.py

```diff
@@ -795,6 +835,36 @@ def _validate_default(enum_type: Any, default: Any, allow_none: bool) -> None:
     EnumField = EnumFieldV3
 
     DictField = m.fields.Dict
+
+    class UnionFieldV3(m.fields.Field):
+        def __init__(self, fields: dict[type, m.fields.Field], **kwargs: Any):
+            self.fields = fields
+            super().__init__(**kwargs)
+
+        def _serialize(self, value: Any, attr: Any, obj: Any, **kwargs: Any) -> Any:
+            errors = []
+            for type, field in self.fields.items():
+                if not isinstance(value, type):
+                    continue
+                try:
+                    return field._serialize(value, attr, obj, **kwargs)
+                except m.ValidationError as e:
+                    errors.append(e.messages)
+            raise m.ValidationError(message=errors, field_name=attr)
+
+        def _deserialize(self, value: Any, attr: Any, data: Any, **kwargs: Any) -> Any:
+            errors = []
+            for type, field in self.fields.items():
+                try:
+                    result = field.deserialize(value, attr, data, **kwargs)
+                    if not isinstance(result, type):
```

Comment:
I agree with such check in `_serialize` method but how it can be deserialized in wring type? and what about `Any` in union like `str | bool | Any` ?  Also `Any | None` will fail but for another reason